### DISCUSSION TITLE
프로젝트 내에서 Issues와 Discussions 이용 방식 제안

### DISCUSSION THREAD

**Discussion by chadseok** [wanted-pre-onboarding-side-project/habit-tracker#28] [Ideas]
## ISSUE
- Issues에 올릴 것들은 **현재 main 코드**에서 생긴 버그, 개선할 점 또는 기타이슈에 관한 것들이다
- 문제점에 대해 작성하고 마지막에 해결책이나 다른방식을 제안을 할 수 있다
- Issue 제목으로는 구어체는 최대한 지양한다(ex. '~~해봐요', '~~어떤가요')

<Br> 

## Discussion
- 프로젝트 운용에 관해 비동기적으로 의견을 주고 받는 곳
- **현재 코드**에서 개선할 점과 관련없는 새로운 기능 추가는 Discussion에 제안을 하고 토론을 한다
- 보이스로 회의를 한 내용들은 Announcement탭에 작성해서 기록을 한다. 
- 회의를 한 내용들 중에 wiki로 기록할 만한 것들 (컨벤션, git flow, 프로토타입 등등)을 wiki에 따로 작성한다

**Comment by greyHairChooseLife** [comment]
저도 커밋 템플릿, pr템플릿에 이어 issue템플릿을 만드는 것에 동의합니다.

#24 에서 동일 주제 논의가 되고 있었는데, discussion으로 옮겨 와 주셔서 감사합니다. 다만 따봉 이모지 남겨주신 것 보니 읽어 보신 듯 한데, 기존에 제안한 내용을 바탕으로 개선/추가/삭제하여 제안 해 주시면 좋을 것 같습니다. 어떻게 합쳐야 하는지, 또는 앞서 말한 내용들 중 무엇을 제외 하자는 것인지 반복해서 생각하게 되네요.

또 제안 해 주신 것들 관련해서 몇 가지 여쭤봅니다.

1. `main 브랜치`에 한정하는 이유가 있을까요? 다른 브랜치는 어떻게 관리하면 좋을까요. 
>현재 main 코드에서 생긴...

2. Announcement탭이란 것이 현재 화면 우상단에 `Category` 말씀하시는게 맞나요?

**Comment by chadseok** [comment]
1. 네 main브랜치라고 한것은 저희가 feature단위로 main에 머지하고 난후 미처 발견하지 못한 문제을 이슈로 남기고 기능개발 중간에 발견을 했다면 바로 해결하거나 논의가 필요하다면 discussion에 남기면 될것같습니다

2. 네 맞습니다

**Reply by greyHairChooseLife** [reply]
그러네요, 생각 해 보니  `main` 외의 브랜치는 `PR`과 `merge`도 한번 뿐일텐데 버그가 있다면 어차피 리뷰 도중에 걸러지겠어요.

**Comment by jiyeon2** [comment]
## ISSUE
- Issues에 올릴 것들은 **현재 main 코드**에서 생긴 버그, 개선할 점 또는 기타이슈에 관한 것들이다
  - 프로젝트 ’코드‘에 변경사항이 생기는 작업 논의
- 문제점에 대해 작성하고 마지막에 해결책이나 다른방식을 제안을 할 수 있다
- Issue 제목으로는 구어체는 최대한 지양한다(ex. '~~해봐요', '~~어떤가요')



<Br> 

## Discussion
- 프로젝트 운용에 관해 비동기적으로 의견을 주고 받는 곳
  - 코드 변경사항이 없는 협업 프로세스 수정, 아이디어 제시 등은 discussion탭
- **현재 코드**에서 개선할 점과 관련없는 새로운 기능 추가는 Discussion에 제안을 하고 토론을 한다
- 보이스로 회의를 한 내용들은 Announcement탭에 작성해서 기록을 한다. 
- 회의를 한 내용들 중에 wiki로 기록할 만한 것들 (컨벤션, git flow, 프로토타입 등등)을 wiki에 따로 작성한다



## 정리

1. 코드에 없는 새로운 기능 : `discussion` :point_right: (구체화 및 개발 확정 시) :point_right: `issue: TODO라벨` :point_right: `project: Todo`
2. 코드에 있는 기능 수정 및 버그 : `issue` :point_right: (이슈에서 논의 후 결정 되는 대로) :point_right: `project: Todo`
3. 협업 프로세스(기존에 있든, 없든) : `(추가/수정)discussion` :point_right: (구체화 및 확정 시) :point_right: `wiki`

## 기대 효과
issue 탭에서는 코드 관련 문제 내역과 이슈만 확인할 수 있다
discussion 탭에서는 팀 내부 의사 결정 과정을 확인할 수 있다


<Br> 

#24 이슈라벨 추가 에서 논의한 내용과 여기서 제안된 내용을 합쳐보면 위와 같습니다
여기 관련해서 다른 의견이 있을까요?
없으면 일단 이 기준으로 진행하고, 수정이 필요하다고 생각되면
다시 논의를 거쳐 규칙을 변경하면 어떨까 싶습니다

**Reply by greyHairChooseLife** [reply]
네, wiki에 관련 내용 확정 지으면서 정리하면 될 것 같습니다.
